File: cplusplus/tests/key_pair.cpp
#include <catch2/catch_test_macros.hpp>

#include "autograph.h"

TEST_CASE("Key pair", "[key_pair]") {
  Autograph::KeyPair emptyKeyPair;

  SECTION("should generate ephemeral key pairs") {
    auto [success, keyPair] = Autograph::generateKeyPair();
    REQUIRE(success == true);
    REQUIRE(keyPair != emptyKeyPair);
  }

  SECTION("should generate identity key pairs") {
    auto [success, keyPair] = Autograph::generateIdentityKeyPair();
    REQUIRE(success == true);
    REQUIRE(keyPair != emptyKeyPair);
  }
}

File: rust/tests/key_pair.rs
use rand::rngs::OsRng;

use autograph::{generate_identity_key_pair, generate_key_pair, KEY_PAIR_SIZE};

#[test]
fn test_generate_ephemeral_key_pair() {
    let key_pair = generate_key_pair(OsRng).unwrap();
    assert_ne!(key_pair, [0; KEY_PAIR_SIZE]);
}

#[test]
fn test_generate_identity_key_pair() {
    let key_pair = generate_identity_key_pair(OsRng).unwrap();
    assert_ne!(key_pair, [0; KEY_PAIR_SIZE]);
}
